src/utils/utils/matchSelector.ts

```diff
@@ -0,0 +1,54 @@
+import {
```

Comment:
please explain this file

Reply:
To understand if ClusterUDN is managing a namespace, I have to go through its `matchLabels` and `matchExpressions` and verify that the project labels verify both. 

The matchExpression that I use with ClusterUDN in the networking plugin is

```
{key: 'kubernetes.io/metadata.name', operator: 'In', values: [projectA, projectB]}
```

But in general ClusterUDN can also have `matchLabels` and different kind of `matchExpressions` so to support ClusterUDN created through command line or yaml I wrote this function 


I created this file in the utils as this kind of validation is [common in k8s](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements) for other things so I think it can be helpful in the future.

Reply:
The k8s field that has `matchLabels` and `matchExpressions` is called commonly `Selector` so this is why I named the file `matchSelector`.

The flow is:

`matchLabels` requirements can be transformed to expressions simply by using the `Equal` operator. So I did that to simplify the function. 

After that I just had to transform operators to functions like `includes` and `! includes`